### PR TITLE
User-attributes in authorisation and failing OData queries

### PR DESCRIPTION
**Pull request by olirogers**
Hi,

I've been trying to enable user authorisation by attributes and there are two scenarios that are failing where CDS is generating incorrect SQL.

This is an example of this issue, and when it works, it can be used as a working example of using user attributes in CAP.

These two scenarios are:
- OData V4 Apply / Group by
- Inline count with expand and filter

More details (and log output) can be found in the [README in the issue folder](https://github.com/olirogers/cap-community/blob/user-attrs/issues/user-attrs/README.md)

Creating this here so that others can replicate the issue.

Thanks,

Oli

### CODE REVIEW

File: issues/user-attrs/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+Note: The documentation (here)[https://cap.cloud.sap/docs/node.js/authentication#authentication-strategies] states to use `userAttributes` in the `package.json` however, only `attributes` property works.
+
+
+## Try it out
+
+First, get the service up and running:
+
+1. Get to right place: `cd issues/user-attrs/`
+2. Install deps and run in memory (with debug): `npm i && DEbUG=true cds run --in-memory`
```
Comment (gregorwolf) [inline issues/user-attrs/README.md]: With the spelling correction:

DEBUG=true

the debug output would work.

**Comment by olirogers** [comment]
Thanks @gregorwolf. I've updated the README, it was just a typo there, debug output is further down the README.

File: issues/user-attrs/README.md (hunk 1 above)
Comment (olirogers) [reply issues/user-attrs/README.md]: Fixed in commit 82808b1

**Comment by vobu** [comment]
sorry for the mess, thought I could get it to merge and help out gregor here. turns out i can't :(

**Pull request merged**